### PR TITLE
Add pixel and intensity functions.

### PR DESCRIPTION
**Pull request by Damdoshi**

### CODE REVIEW

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp [hunk 1]
```diff
@@ -37,6 +34,16 @@ void LedMatrixMax7219::sendBytesToMax7219(byte data)
    }
 }
 
+void LedMatrixMax7219::setIntensity(byte a, byte b, byte c, byte d)
```
Comment (monpierrelisa) [inline sources/led_matrix_max7219/ledMatrixMax7219.cpp:37]: Please add a real name for these variables

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp [hunk 2]
```diff
@@ -24,6 +20,7 @@ void LedMatrixMax7219::initPins()
   pinMode(clockPin, OUTPUT);
   pinMode(chipSelectPin, OUTPUT);
   pinMode(dataInPin, OUTPUT);
+  delay(500);
```
Comment (monpierrelisa) [inline sources/led_matrix_max7219/ledMatrixMax7219.cpp:23]: delay 50 is sufficient normally no ?

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp [hunk 3]
```diff
@@ -52,6 +59,7 @@ void LedMatrixMax7219::initMax7219Component()
   writeToRegister(scanLimitRegister,   scan8LEDs);
   writeToRegister(shutdownRegister,    normalMode);
   writeToRegister(displayTestRegister, displayMode);
+  delay(500);
```
Comment (monpierrelisa) [inline sources/led_matrix_max7219/ledMatrixMax7219.cpp:62]: delay 50 is sufficient normally no ?

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp [hunk 4]
```diff
@@ -70,28 +78,59 @@ void LedMatrixMax7219::endWriteLine()
...
 
-void LedMatrixMax7219::putPixel(byte indexDigit, byte indexSegment, byte pixel)
+void LedMatrixMax7219::setPixel(byte x, byte y, byte color)
 {
-  matrix[indexDigit][indexSegment] = pixel;
+  byte bloc = x >> 3;
+  byte pixel = x - (bloc << 3);
+  byte *target = &matrix[y][bloc];
+
+  if (color)
```
Comment (monpierrelisa) [inline sources/led_matrix_max7219/ledMatrixMax7219.cpp:112]: add {}

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp [hunk 5]
```diff
@@ -70,28 +78,59 @@ void LedMatrixMax7219::endWriteLine()
...
+{
+  byte bloc = x >> 3;
+  byte pixel = x - (bloc << 3);
+  const byte *target = &matrix[y][bloc];
+
+  return ((*target >> pixel) & 1);
+}
+
+byte LedMatrixMax7219::getWidth() const
+{
```
Comment (monpierrelisa) [inline sources/led_matrix_max7219/ledMatrixMax7219.cpp:128]: move these getter in .h to become inlined function

**Review by monpierrelisa — CHANGES_REQUESTED** [review]

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp (hunk 2 above)
Comment (Damdoshi) [reply sources/led_matrix_max7219/ledMatrixMax7219.cpp:23]: The init sequence that allow the card to be well reseted use 500ms as delay in those places.

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp (hunk 3 above)
Comment (Damdoshi) [reply sources/led_matrix_max7219/ledMatrixMax7219.cpp:62]: The init sequence that allow the card to be well reseted use 500ms as delay in those places.

File: sources/led_matrix_max7219/ledMatrixMax7219.cpp (hunk 1 above)
Comment (Damdoshi) [reply sources/led_matrix_max7219/ledMatrixMax7219.cpp:37]: intensity_level_for_matrix_a ?

**Comment by Damdoshi** [comment]
Changes are done.

**Review by monpierrelisa — APPROVED** [review]

**Pull request merged**